source/rock/frontend/AstPrinter.ooc

```diff
@@ -0,0 +1,370 @@
+import io/Writer
+import structs/[List, ArrayList]
+
+import rock/io/TabbedWriter
+
+import rock/middle/[Addon, AddressOf, Argument, ArrayAccess, ArrayCreation, ArrayLiteral,
+    BaseType, BinaryOp, Block, BoolLiteral, CallChain, Cast, CharLiteral,
+    ClassDecl, CommaSequence, Comparison, Conditional, ControlStatement,
+    CoverDecl, Declaration, Dereference, Else, EnumDecl, Expression,
+    FloatLiteral, FlowControl, Foreach, FunctionCall, FunctionDecl,
+    FuncType, If, Import, Include, InterfaceDecl, InterfaceImpl, IntLiteral,
+    Line, Literal, Match, Module, NamespaceDecl, Node, NullLiteral, OperatorDecl,
+    Parenthesis, PropertyDecl, RangeLiteral, Return, SafeNavigation, Scope,
+    Statement, StringLiteral, StructLiteral, TemplateDef, Ternary, Try, Tuple,
+    Type, TypeDecl, TypeList, UnaryOp, Use, UseDef, VariableAccess, VariableDecl,
+    Version, Visitor, While]
```

Comment:
Yeah, the AST being split into a file for each Node type leads to this kind of thing unfortunately :P